packages/ckb/src/spore/leap.ts

```diff
@@ -0,0 +1,270 @@
+import { BtcTimeCellsParams, RgbppCkbVirtualTx } from '../types/rgbpp';
+import { append0x, calculateTransactionFee } from '../utils';
+import {
+  btcTxIdFromBtcTimeLockArgs,
+  buildSpvClientCellDep,
+  calculateCommitment,
+  compareInputs,
+  genBtcTimeLockScript,
+  lockScriptFromBtcTimeLockArgs,
+  transformSpvProof,
+} from '../utils/rgbpp';
+import {
+  Hex,
+  LeapSporeFromBtcToCkbVirtualTxParams,
+  LeapSporeFromCkbToBtcVirtualTxParams,
+  SporeLeapVirtualTxResult,
+} from '../types';
+import {
+  BTC_JUMP_CONFIRMATION_BLOCKS,
+  RGBPP_TX_WITNESS_MAX_SIZE,
+  RGBPP_WITNESS_PLACEHOLDER,
+  getBtcTimeLockConfigDep,
+  getBtcTimeLockDep,
+  getRgbppLockConfigDep,
+  getRgbppLockDep,
+  getRgbppLockScript,
+  getSporeTypeDep,
+} from '../constants';
+import { generateSporeTransferCoBuild } from '../utils/spore';
+import { NoRgbppLiveCellError, RgbppUtxoBindMultiSporesError } from '../error';
+import {
+  addressToScript,
+  getTransactionSize,
+  serializeOutPoint,
+  serializeScript,
+  serializeWitnessArgs,
+} from '@nervosnetwork/ckb-sdk-utils';
+import { blockchain } from '@ckb-lumos/base';
+import { buildBtcTimeUnlockWitness } from '../rgbpp';
+
+/**
+ * Generate the virtual ckb transaction for leaping spore from BTC to CKB
+ * @param collector The collector that collects CKB live cells and transactions
+ * @param sporeRgbppLockArgs The spore rgbpp cell lock script args whose data structure is: out_index | bitcoin_tx_id
+ * @param sporeTypeBytes The spore type script serialized bytes
+ * @param toCkbAddress The receiver ckb address
+ * @param witnessLockPlaceholderSize The WitnessArgs.lock placeholder bytes array size and the default value is 5000
+ * @param ckbFeeRate The CKB transaction fee rate, default value is 1100
+ */
+export const genLeapSporeFromBtcToCkbVirtualTx = async ({
+  collector,
+  sporeRgbppLockArgs,
+  sporeTypeBytes,
+  toCkbAddress,
+  isMainnet,
+  witnessLockPlaceholderSize,
+  ckbFeeRate,
+}: LeapSporeFromBtcToCkbVirtualTxParams): Promise<SporeLeapVirtualTxResult> => {
+  const sporeRgbppLock = {
+    ...getRgbppLockScript(isMainnet),
+    args: append0x(sporeRgbppLockArgs),
+  };
+  const sporeCells = await collector.getCells({ lock: sporeRgbppLock, isDataEmpty: false });
```

Comment:
```suggestion
  const sporeCells = await collector.getCells({ lock: sporeRgbppLock });
```

Should search all cells locked by the expected rgbpp-lock, including empty data cells.

Reply:
Does Spore protocol allow cell data to be empty? If allowed, it does need to be added

Reply:
I think it's possible that an empty data cell could be bound to the same rgbpp-lock with spore a spore cell.

### All the rgbpp asset cells that are isomorphic with a specific BTC_UTXO, should be unlocked in one single transaction.

Reply:
After discussion, https://github.com/ckb-cell/rgbpp-sdk/pull/130/commits/529e68c1a1593c1e711d90d30f2d617851105509 renamed `isDataEmpty ` to `isDataMustBeEmpty`, which is more clear.